src/targets/gpu/gemm.cpp

```diff
@@ -171,6 +171,35 @@ shape miopen_gemm::compute_shape(const std::vector<shape>& inputs) const
 {
     std::vector<shape> input_shapes(inputs.begin(), inputs.begin() + inputs.size() - 1);
     check_shapes{input_shapes}.not_broadcasted();
+    auto a_strides = inputs[0].strides();
+    auto dim_0     = a_strides.size() - 2;
+    if(a_strides.size() > 2)
+    {
+        if(!std::all_of(a_strides.begin(), a_strides.begin() + dim_0, [&](auto batch_size) {
+               return std::all_of(a_strides.begin() + dim_0, a_strides.end(), [&](auto data_size) {
```

Comment:
This doesnt seem to check for transposed batches. It would be better to get the max stride of the last two dimensions and then do `adjacent_find` to make sure each stride is greater than the max stride and greater than its previous stride.